### PR TITLE
Scripts

### PR DESCRIPTION
**Pull request by allanchau**
This PR updates the scripts tag to support data urls and some additional attribute that were previously missing.

**Related PRs*

- #233 

**Testing**

- [x] Start the test app and search the page source for `<script data-test="test"></script>`.

**Notes**

- I've updated the dev environment to start with `yarn start` instead of `yarn dev`.
- The reset/import path has been updated to `/reset`. The default route `/` doesn't run any scripts anymore.

### CODE REVIEW

File: docs/defaults.rst [hunk 1]
```diff
@@ -156,6 +156,24 @@ You should use the existing array as the array that is resolved with the promise
 
 The script objects can contain an additional ``inHead`` boolean option to optionally load the script in the head tag.
 
+To create data attributes, you can add a ``dataAttributes`` property with a key being prefixed with ``data-``. For example::
```
Comment (smebberson) [inline docs/defaults.rst]: Please update to `...property with a key that will be prefixed with ``data-`` when output in HTML.`

File: lib/scripts.js [hunk 2]
```diff
@@ -1,5 +1,29 @@
 'use strict';
 
+const parseDataAttributes = (scriptObj) => {
+
+    // Don't mutate the original object.
+    const script = Object.assign({}, scriptObj);
+
+    if (!script.dataAttributes) {
```
Comment (smebberson) [inline lib/scripts.js]: I think this should be above 6. You're actually, needlessly, duplicating and returning a different object by doing it this way.

File: lib/scripts.js [hunk 3]
```diff
@@ -1,5 +1,29 @@
...
+        return script;
+    }
+
+    const dataAttributes = Object.keys(script.dataAttributes);
```
Comment (smebberson) [inline lib/scripts.js]: Can't this be written more cleanly using `map`?

File: test/lib-scripts.test.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const {
+    parseDataAttributes,
+    parseScripts,
+} = require('../lib/scripts');
+
+test('it does not mutate the original script', () => {
```
Comment (smebberson) [inline test/lib-scripts.test.js]: You could put another test here if you want to make sure it doesn't needlessly duplicate the object, kinda worthless, but would be interesting to validate my first comment.

File: views/partials/foot.jade [hunk 5]
```diff
@@ -2,9 +2,9 @@ if scripts && scripts.length
 	each script in scripts
 		if !script.inHead
 			if script.content
-					script(async=script.async crossorigin=script.crossorigin defer=script.defer integrity=script.integrity src=script.src text=script.text type=script.type)!= script.content
+					script(async=script.async crossorigin=script.crossorigin defer=script.defer integrity=script.integrity nomodule=script.nomodule nonce=script.nonce referrerpolicy=script.referrerpolicy src=script.src text=script.text type=script.type)&attributes(script.dataAttributes || {})!= script.content
```
Comment (smebberson) [inline views/partials/foot.jade]: I would prefer the `script.dataAttributes || {}` happen in JavaScript, not Jade. This will make it easier to move away from Jade.

File: views/partials/head.jade [hunk 6]
```diff
@@ -14,6 +14,6 @@ head
         each script in scripts
             if script.inHead
                 if script.content
-                    script(async=script.async crossorigin=script.crossorigin defer=script.defer integrity=script.integrity src=script.src text=script.text type=script.type)!= script.content
+                    script(async=script.async crossorigin=script.crossorigin defer=script.defer integrity=script.integrity nomodule=script.nomodule nonce=script.nonce referrerpolicy=script.referrerpolicy src=script.src text=script.text type=script.type)&attributes(script.dataAttributes || {})!= script.content
```
Comment (smebberson) [inline views/partials/head.jade]: Same as above regarding `dataAttributes`...

File: lib/scripts.js (hunk 3 above)
Comment (allanchau) [reply lib/scripts.js]: `dataAttributes` and `parsedAttributes` are objects though, I'm not sure how we can use `map` without converting them between objects and arrays a few times.

I could just remove the `dataAttributes` variable and use `Object.keys(script.dataAttributes)` directly if that makes it easier to read?

File: lib/scripts.js (hunk 2 above)
Comment (allanchau) [reply lib/scripts.js]: @smebberson With the Jade change below I think we need to keep this here and add a `script.dataAttributes` default.

File: lib/util/parse.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    if (!newObj.dataAttributes) {
+
+        newObj.dataAttributes = {};
+
+        return newObj;
+
+    }
+
+    const dataAttributes = Object.keys(newObj.dataAttributes);
```
Comment (smebberson) [inline lib/util/parse.js]: @allanchau, how come you're not using `map` here directly on `newObj.dataAttributes`?

Comment (allanchau) [reply lib/util/parse.js]: @smebberson oh sorry, I might have accidentally closed the other convo. If we use map we would have to convert the array to an object later on anyway, seems like that would be a bit more complex than what I've done here?

Comment (smebberson) [reply lib/util/parse.js]: @allanchau, yeah, fair enough, makes sense.

File: lib/scripts.js (hunk 2 above)
Comment (smebberson) [reply lib/scripts.js]: Yep, makes sense.
